syntax = "proto3";

package Proto;

// 
message Request {

	// 包类型
	enum Type {
		LOGIN	= 0;
		LOGOUT	= 1;
		SENDMSG = 2;
	}
	Type type = 1;
	oneof pdu {
		Request_login	request_login	= 11;
		Request_logout	request_logout	= 12;
		Request_sendmsg request_sendmsg	= 13;
	}
}

// 服务端对客户端的响应
message Response {
	
	// 类型
	enum Type {
		LOGIN	= 0;
		LOGOUT	= 1;
		SENDMSG = 2;
		MESSAGE = 3;
	}

	Type type = 1;
	oneof pdu {
		Response_login	response_login		= 11;
		//Response_logout	response_logout	= 12;
		Response_sendmsg response_sendmsg	= 13;
		//其他账号发送的消息
		Message msg							= 14;
	}
}



//登录请求
message Request_login {
	uint64 ID		= 1;
	string password = 2;
	// Next available id: 1
}

//登录应答
message Response_login {
	
	// 是否登录成功
	enum STAT {
		SUCCESS        = 0;
		ACCOUNT_NULL   = 1;
		ACCOUNT_LOCK   = 2;
		PASSWORD_ERROR = 3;
		ERROR		   = 10;
	}

	STAT stat = 1;
	string friends = 2;
	// Next available id: 1
}

//登出请求
message Request_logout {
	uint64 ID = 1;
	// Next available id: 1
}

////登出应答
//message Response_logout {
//	uint64 ID = 1;
//	// Next available id: 1
//}


//请求发送消
message Request_sendmsg {
	uint32 msgID	= 1; //消息ID
	uint64 ID		= 2; //自己的ID
	uint64 objID	= 3; //对方的ID
	string msg		= 4;

	// Next available id: 1
}

//请求发送消
message Response_sendmsg {
	uint32 msgID	= 1;
	bool stat		= 2;
	// Next available id: 1
}

// 消息推送
message Message {
	// 消息类型
	//enum MsgType {

	//}

	//MsgType type	=1;
	uint64 fromID	=2;
	string msg		=3;
}
